pkg/common-controller/snapshot_controller_base.go

```diff
@@ -218,6 +218,15 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 			if err == nil {
 				klog.V(5).Infof("passed checkAndUpdateSnapshotClass for snapshot %q", key)
 				ctrl.updateSnapshot(newSnapshot)
+			} else if newSnapshot.ObjectMeta.DeletionTimestamp != nil && errors.IsNotFound(err) {
+				// If the VolumeSnapshotClass is not found, we still need to process an update
```

Comment:
If this is a delete event, can we let the logic on line 256 to handle it?
https://github.com/kubernetes-csi/external-snapshotter/pull/275/files#diff-77f790654ab434b7aac6ae9b88ecd4b2R256

Reply:
The issue that I've seen here is that the VolumeSnapshot would be queued for deletion, but the VolumeSnapshotClass would be deleted before all the associated objects (PVC/VolumeSnapshotContent) are removed from the cluster.

In this event, the snapshot still exists in the cluster, so it falls into the condition at line 214. There was an error retrieving the class from line 218, and then we exited the loop at line 231 only to repeat this again. I updated the conditions at line 255 so that this could be handled if it's caught in time, and if not it'll get deleted during the next update.

Reply:
How about add a check for `if newSnapshot.ObjectMeta.DeletionTimestamp != nil && errors.IsNotFound(err)`.  Only `return false` on line 231 if the check is false.  This way it can be handled by the logic on line 256.

Reply:
I updated the current check to include this. This should still work, and I haven't been able to reproduce it after running ~20 times (previously it would manifest ~5-10% of the time).